src/number_theory/legendre_symbol/jacobi_symbol.lean

```diff
@@ -201,6 +201,50 @@ end
 lemma mod_left' {a₁ a₂ : ℤ} {b : ℕ} (h : a₁ % b = a₂ % b) : J(a₁ | b) = J(a₂ | b) :=
 by rw [mod_left, h, ← mod_left]
 
+/-- If `p` is prime, `J(a | p) = -1` and `p` divides `x^2 - a*y^2`, then `p` must divide
+`x` and `y`. -/
+lemma prime_dvd_of_eq_neg_one {p : ℕ} [fact p.prime] {a : ℤ} (h : J(a | p) = -1)
+  {x y : ℤ} (hxy : ↑p ∣ x ^ 2 - a * y ^ 2) : ↑p ∣ x ∧ ↑p ∣ y :=
+begin
+  rw [← legendre_sym.to_jacobi_sym] at h,
+  exact legendre_sym.prime_dvd_of_eq_neg_one h hxy,
+end
+
+/-- We can pull out a product over a list in the first argument of the Jacobi symbol. -/
+lemma mul_left_list {l : list ℤ} {n : ℕ} :
```

Comment:
I'm not great at naming lemmas but somehow I feel like `prod` should feature in the name of this one (and not `mul` because there is no `*`). Same for the next lemma.

Reply:
OK; I think I 'll go for `list_prod_left` and `list_prod_right`.